Can't compile on M1 Mac

```
➜  sslscan git:(master) ✗ make static &> output
➜  sslscan git:(master) ✗ tail -30  output
      _showCertificate in sslscan-450c45.o
  "_X509_getm_notAfter", referenced from:
      _checkCertificate in sslscan-450c45.o
      _showCertificate in sslscan-450c45.o
  "_X509_getm_notBefore", referenced from:
      _checkCertificate in sslscan-450c45.o
      _showCertificate in sslscan-450c45.o
  "_X509_signature_print", referenced from:
      _showCertificate in sslscan-450c45.o
  "_X509_verify_cert_error_string", referenced from:
      _showCertificate in sslscan-450c45.o
  "_X509v3_get_ext", referenced from:
      _checkCertificate in sslscan-450c45.o
  "_d2i_OCSP_RESPONSE", referenced from:
      _ocsp_resp_cb in sslscan-450c45.o
  "_d2i_PKCS12_fp", referenced from:
      _loadCerts in sslscan-450c45.o
  "_i2a_ASN1_INTEGER", referenced from:
      _ocsp_certid_print in sslscan-450c45.o
  "_i2a_ASN1_OBJECT", referenced from:
      _ocsp_resp_cb in sslscan-450c45.o
      _ocsp_certid_print in sslscan-450c45.o
      _showCertificate in sslscan-450c45.o
  "_i2a_ASN1_STRING", referenced from:
      _ocsp_resp_cb in sslscan-450c45.o
      _ocsp_certid_print in sslscan-450c45.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[1]: *** [sslscan] Error 1
make: *** [static] Error 2
```

If any additional details would be helpful just let me know! In the meantime I'll just use Linux lol

Progress! Now I get this:

```
ld: warning: ignoring file crypto/x509v3/v3_genn.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_int.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_ia5.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_ncons.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_lib.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_info.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_pcons.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_pmaps.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_pcia.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_pci.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_prn.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_skey.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_pku.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_purp.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_tlsf.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_sxnet.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3err.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
ld: warning: ignoring file crypto/x509v3/v3_utl.o, building for macOS-arm64 but attempting to link with file built for unknown-x86_64
Undefined symbols for architecture arm64:
  "_BN_num_bits_word", referenced from:
      _bn_div_words in bn_asm.o
  "_CRYPTO_cbc128_decrypt", referenced from:
      _Camellia_cbc_encrypt in cmll_cbc.o
  "_CRYPTO_cbc128_encrypt", referenced from:
      _Camellia_cbc_encrypt in cmll_cbc.o
  "_Camellia_decrypt", referenced from:
      _Camellia_cbc_encrypt in cmll_cbc.o
  "_Camellia_encrypt", referenced from:
      _Camellia_cbc_encrypt in cmll_cbc.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libcrypto.dylib] Error 1
make[1]: *** [all] Error 2
make: *** [openssl/libcrypto.a] Error 2
```

The warnings go on and on, all along the same lines.

--- a/Makefile
+++ b/Makefile
@@ -9,6 +9,7 @@ endif
 
 # Detect OS
 OS := $(shell uname)
+ARCH := $(shell uname -m)
 
 # Handle different version of Make
 ifeq ($(OS), SunOS)
@@ -61,7 +62,11 @@ ifeq ($(STATIC_BUILD), TRUE)
 PWD          = $(shell pwd)/openssl
 LDFLAGS      += -L${PWD}/
 CFLAGS       += -I${PWD}/include/ -I${PWD}/
+ifeq ($(OS), Darwin)
+LIBS	     = ./openssl/libssl.a ./openssl/libcrypto.a -lz -lpthread
+else
 LIBS         = -lssl -lcrypto -lz -lpthread
+endif
 ifneq ($(OS), FreeBSD)
 	LIBS += -ldl
 endif
@@ -129,8 +134,13 @@ opensslpull:
 
 # Need to build OpenSSL differently on OSX
 ifeq ($(OS), Darwin)
+ifeq ($(ARCH), arm64)
+OSSL_TARGET=darwin64-arm64-cc
+else
+OSSL_TARGET=darwin64-x86_64-cc
+endif
 openssl/Makefile: .openssl.is.fresh
-	cd ./openssl; ./Configure -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIC enable-weak-ssl-ciphers zlib darwin64-x86_64-cc
+	cd ./openssl; ./Configure -fstack-protector-all -D_FORTIFY_SOURCE=2 -fPIC enable-weak-ssl-ciphers zlib $(OSSL_TARGET)
 # Any other *NIX platform
 else
 openssl/Makefile: .openssl.is.fresh

